### PR TITLE
Backend/feat/Storage parser

### PR DESCRIPTION
**Pull request by ErasedSoul**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Storage Parser

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/alchemist/src/Alchemist/Utils.hs [hunk 1]
```diff
@@ -1,9 +1,48 @@
...
+  let extractWords = splitWhen (`elem` typeDelimiter) <$> fieldTypes
+   in filter (not . null) $ concatMap (map makeTypeQualified) (nub extractWords)
+
+findMatchingSqlType :: String -> Maybe String
+findMatchingSqlType haskellType =
+  case filter ((haskellType =~) . fst) defaultSQLTypes of
+    [] -> Nothing
+    ((_, sqlType) : _) -> Just sqlType
+
+defaultSQLTypes :: [(String, String)]
```
Comment (hkmangla) [inline Backend/app/alchemist/src/Alchemist/Utils.hs:34]: This can go to Storage Parse only.

File: Backend/app/alchemist/src/Alchemist/Utils.hs [hunk 2]
```diff
@@ -1,9 +1,48 @@
...
+    ("TimeOfDay", "UTCTime")
+  ]
+
+getQualifiedImport :: String -> Maybe String
+getQualifiedImport = \case
+  "Text" -> Just "Data.Text.Text"
+  "Maybe" -> Just "Data.Maybe.Maybe"
+  "Id" -> Just "Beckn.Types.Id"
+  "TimeOfDay" -> Just "Beckn.Types.TimeOfDay"
+  "Int" -> Just "Int"
```
Comment (hkmangla) [inline Backend/app/alchemist/src/Alchemist/Utils.hs:47]: Isn't it `Data.Int`

File: Backend/app/alchemist/src/Alchemist/Utils.hs [hunk 3]
```diff
@@ -1,9 +1,48 @@
...
+defaultSQLTypes =
+  [ ("Text", "CHARACTER(36)"),
+    ("Id ", "Varchar(50)"),
+    ("TimeOfDay", "UTCTime")
+  ]
+
+getQualifiedImport :: String -> Maybe String
+getQualifiedImport = \case
+  "Text" -> Just "Data.Text.Text"
+  "Maybe" -> Just "Data.Maybe.Maybe"
```
Comment (hkmangla) [inline Backend/app/alchemist/src/Alchemist/Utils.hs:44]: One Maybe is extra.

File: Backend/app/alchemist/src/Alchemist/Utils.hs [hunk 4]
```diff
@@ -1,9 +1,48 @@
...
+  where
+    replaceOrKeep :: String -> String
+    replaceOrKeep word = fromMaybe (if '.' `elem` word || word `elem` ["", ")", "(", " "] then word else error "Type not determined") (getQualifiedImport word)
+
+figureOutImports :: [String] -> [String]
+figureOutImports fieldTypes =
+  let extractWords = splitWhen (`elem` typeDelimiter) <$> fieldTypes
+   in filter (not . null) $ concatMap (map makeTypeQualified) (nub extractWords)
+
+findMatchingSqlType :: String -> Maybe String
```
Comment (hkmangla) [inline Backend/app/alchemist/src/Alchemist/Utils.hs:28]: Sql related stuff can go to storage parser only.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/alchemist/src/Alchemist/Utils.hs (hunk 3 above)
Comment (ErasedSoul) [reply Backend/app/alchemist/src/Alchemist/Utils.hs:44]: Screenshot 2023-11-23 at 4 30 09 PM

Type will also be coverted to fulll name. 
Maybe is the constructor name defined in module Data.Maybe

**Review by hkmangla — APPROVED** [review]

**Pull request merged**